modules/mobile/spec/models/adapters/appointments_vaos_v2_adapter_spec.rb

```diff
@@ -3,30 +3,42 @@
 require 'rails_helper'
 
 describe Mobile::V0::Adapters::VAOSV2Appointments, :aggregate_failures do
-  # while hashes will work for these tests, this better reflects the data returned from the VAOS service
   def appointment_data(index = nil)
     appts = index ? raw_data[index] : raw_data
     Array.wrap(appts).map { |appt| OpenStruct.new(appt) }
   end
 
-  def appointment_by_id(id)
-    raw_data.find { |appt| appt[:id] == id }
-  end
-
-  def parse_appointment(appt)
-    subject.parse(Array.wrap(appt)).first
+  def appointment_by_id(id, overrides: {}, without: [])
```

Comment:
This is a great way to manage test data. Wonder if we can make this more generic and move it to the mobile rails_helper as we create more unit tests rather then request specs.

Reply:
It's plausible, but I doubt most of the unit tests will be like this. I think this is good enough for the current purposes, but I'm not sure it's robust enough for more complex test cases. Definitely something to keep in mind if we keep finding ourselves in similar situations.